### ISSUE TITLE
SnapshotLifecycleRestIT testFullPolicySnapshot failure 

### ISSUE THREAD

**Issue by davidkyle** [elastic/elasticsearch#50358]
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-windows-compatibility/os=windows-2016/331/console
https://gradle-enterprise.elastic.co/s/fmz3orwsf6ceu

```
org.elasticsearch.client.ResponseException: 
method [DELETE], host [http://127.0.0.1:60674], URI [/_snapshot/my-repo/snap-dgpzyeujtlil-pxstwwurg], status line [HTTP/1.1 503 Service Unavailable]
{"error":{"root_cause":[{"type":"concurrent_snapshot_execution_exception","reason":"[my-repo:snap-dgpzyeujtlil-pxstwwurg/5U_JWp2pRceMG2VKwqteqQ] another snapshot is currently running cannot delete"}],"type":"concurrent_snapshot_execution_exception","reason":"[my-repo:snap-dgpzyeujtlil-pxstwwurg/5U_JWp2pRceMG2VKwqteqQ] another snapshot is currently running cannot delete"},"status":503}

at __randomizedtesting.SeedInfo.seed([A77F6987797E068:3AE5B97574CB83D]:0)
at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:283)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:261)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
at org.elasticsearch.client.RestClient.performRequest(RestClient.java:235)
at org.elasticsearch.test.rest.ESRestTestCase.wipeSnapshots(ESRestTestCase.java:613)
at org.elasticsearch.test.rest.ESRestTestCase.lambda$wipeCluster$3(ESRestTestCase.java:504)
at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:854)
at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:503)
at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:287)
...
```

Does not reproduce:
```
gradlew ':x-pack:plugin:ilm:qa:multi-node:integTestRunner' --tests "org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot" \
  -Dtests.seed=A77F6987797E068 \
  -Dtests.security.manager=true \
  -Dtests.locale=ca \
  -Dtests.timezone=Europe/Ulyanovsk \
  -Dcompiler.java=13
```

**Comment by davidkyle** [comment]
A handful of similar failures:

https://gradle-enterprise.elastic.co/s/scv4m7g4w7kcs
https://gradle-enterprise.elastic.co/s/ttguod4m35fqs

Muted on master in ae90f8a65bc068fe24f93a3b4be1a5e4c93cd46a

**Comment by danielmitterdorfer** [comment]
We have another failure in 7.x ([CI link](https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-unix-compatibility/os=opensuse-15-1&&immutable/507/), [build scan](https://gradle-enterprise.elastic.co/s/q7qvnrzbyefb6)):

The symptoms look a bit different to me though:

```
java.lang.AssertionError: 
Expected: a string starting with "snap-"
     but: was "snapuyytvuakl-31toqfartrwfolzgmlvdra"
	at __randomizedtesting.SeedInfo.seed([D4BE8BA449B8A835:DD6726AB6963F060]:0)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.lambda$testFullPolicySnapshot$1(SnapshotLifecycleRestIT.java:110)
```

reproduction line:

```
./gradlew ':x-pack:plugin:ilm:qa:multi-node:integTestRunner' --tests "org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot" -Dtests.seed=D4BE8BA449B8A835 -Dtests.security.manager=true -Dtests.locale=ko-KR -Dtests.timezone=Canada/Central -Dcompiler.java=13 -Druntime.java=8
```

Does not reproduce locally.

**Comment by hendrikmuhs** [comment]
Sorry, I have to re-open:

```
14:07:10 org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT > testFullPolicySnapshot FAILED
14:07:10     java.lang.AssertionError: 
14:07:10     Expected: a string starting with "snap-"
14:07:10          but: was "snapyuqttxlpu-zwgyb5vytjics8kuqznzpq"
14:07:10         at __randomizedtesting.SeedInfo.seed([C2B5BA3D5616DEE8:CB6C173276CD86BD]:0)
14:07:10         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
14:07:10         at org.junit.Assert.assertThat(Assert.java:956)
14:07:10         at org.junit.Assert.assertThat(Assert.java:923)
14:07:10         at org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.lambda$testFullPolicySnapshot$1(SnapshotLifecycleRestIT.java:110)
14:07:10         at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:904)
14:07:10         at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:877)
14:07:10         at org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot(SnapshotLifecycleRestIT.java:101)
14:07:26 REPRODUCE WITH: ./gradlew ':x-pack:plugin:ilm:qa:multi-node:integTestRunner' --tests "org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot" -Dtests.seed=C2B5BA3D5616DEE8 -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=SystemV/MST7MDT -Dcompiler.java=13 -Druntime.java=8
14:07:26 
14:07:26 Suite: Test class org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT
14:07:26   1> [2020-01-17T05:06:57,560][INFO ][o.e.x.s.SnapshotLifecycleRestIT] [testFullPolicySnapshot] before test
14:07:26   1> [2020-01-17T05:06:57,563][INFO ][o.e.x.s.SnapshotLifecycleRestIT] [testFullPolicySnapshot] initializing REST clients against [http://[::1]:39063, http://127.0.0.1:33047, http://[::1]:35411, http://127.0.0.1:37777, http://[::1]:42741, http://127.0.0.1:46725, http://[::1]:35935, http://127.0.0.1:44347]
14:07:26   1> [2020-01-17T05:07:09,867][INFO ][o.e.x.s.SnapshotLifecycleRestIT] [testFullPolicySnapshot] after test
14:07:26   2> REPRODUCE WITH: ./gradlew ':x-pack:plugin:ilm:qa:multi-node:integTestRunner' --tests "org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot" -Dtests.seed=C2B5BA3D5616DEE8 -Dtests.security.manager=true -Dtests.locale=es-DO -Dtests.timezone=SystemV/MST7MDT -Dcompiler.java=13 -Druntime.java=8
14:07:26   2> java.lang.AssertionError: 
14:07:26     Expected: a string starting with "snap-"
14:07:26          but: was "snapyuqttxlpu-zwgyb5vytjics8kuqznzpq"
14:07:26         at __randomizedtesting.SeedInfo.seed([C2B5BA3D5616DEE8:CB6C173276CD86BD]:0)
14:07:26         at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
14:07:26         at org.junit.Assert.assertThat(Assert.java:956)
14:07:26         at org.junit.Assert.assertThat(Assert.java:923)
14:07:26         at org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.lambda$testFullPolicySnapshot$1(SnapshotLifecycleRestIT.java:110)
14:07:26         at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:904)
14:07:26         at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:877)
14:07:26         at org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot(SnapshotLifecycleRestIT.java:101)
```

from 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-unix-compatibility/os=amazon/513/consoleFull

**Comment by droberts195** [comment]
It's now failing in a different way, but immediately after the code that was changed in #51386 to try and fix this:

```
Expected: <[test]>
     but: was <[index-ypfbpkpzll]>
```

That is coming from line 123 of `SnapshotLifecycleRestIT.java`, which is immediately after the code change of #51386:

```
        at __randomizedtesting.SeedInfo.seed([D1426930D73DDCAE:D89BC43FF7E684FB]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.lambda$testFullPolicySnapshot$2(SnapshotLifecycleRestIT.java:123)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:905)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:878)
        at org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot(SnapshotLifecycleRestIT.java:108)
```

This has happened a few times on the 7.x branch.  A couple of examples are:

* https://gradle-enterprise.elastic.co/s/7xml4sxuvqij2
* https://gradle-enterprise.elastic.co/s/2se3277cay4sa

A repro command is:

```
./gradlew ':x-pack:plugin:ilm:qa:multi-node:integTestRunner' --tests "org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot" \
  -Dtests.seed=D1426930D73DDCAE \
  -Dtests.security.manager=true \
  -Dtests.locale=vi-VN \
  -Dtests.timezone=Africa/Maputo \
  -Dcompiler.java=13
```

This didn't reproduce locally for me.

**Comment by davidkyle** [comment]
It came back 😱 

```
org.elasticsearch.client.ResponseException: method [DELETE], host [http://127.0.0.1:38055], URI [/_snapshot/full-policy-repo/snap-fy0vwyhpsrmv9f1bgahmtg], status line [HTTP/1.1 503 Service Unavailable]
{"error":{"root_cause":[{"type":"concurrent_snapshot_execution_exception","reason":"[full-policy-repo:snap-fy0vwyhpsrmv9f1bgahmtg/N6ULZf9_Spiwz8IGF0jHdw] another snapshot is currently running cannot delete"}],"type":"concurrent_snapshot_execution_exception","reason":"[full-policy-repo:snap-fy0vwyhpsrmv9f1bgahmtg/N6ULZf9_Spiwz8IGF0jHdw] another snapshot is currently running cannot delete"},"status":503}
	at __randomizedtesting.SeedInfo.seed([8453CDD48058A5E9:8D8A60DBA083FDBC]:0)
	at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:283)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:261)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:235)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeSnapshots(ESRestTestCase.java:622)
	at org.elasticsearch.test.rest.ESRestTestCase.lambda$wipeCluster$3(ESRestTestCase.java:510)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:875)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:509)
	at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:293)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=sles-15&&immutable/636/console

https://gradle-enterprise.elastic.co/s/l3ntz5zhjvxry/tests/ao2nxxbijves6-3ndcchqutszas

```
 ./gradlew ':x-pack:plugin:ilm:qa:multi-node:integTestRunner' --tests "org.elasticsearch.xpack.slm.SnapshotLifecycleRestIT.testFullPolicySnapshot" \
  -Dtests.seed=8453CDD48058A5E9 \
  -Dtests.security.manager=true \
  -Dtests.locale=ar-SA \
  -Dtests.timezone=SystemV/HST10 \
  -Dcompiler.java=13
```

**Comment by davidkyle** [comment]
Muted on master in 340fcd1

**Comment by cbuescher** [comment]
This failure on 7.7 looks similar.
```
org.elasticsearch.client.ResponseException: method [DELETE], host [http://[::1]:41150], URI [/_snapshot/full-policy-repo/snap-0zx3ektksugws-extubuza], status line [HTTP/1.1 503 Service Unavailable]
{"error":{"root_cause":[{"type":"concurrent_snapshot_execution_exception","reason":"[full-policy-repo:snap-0zx3ektksugws-extubuza/6v6TuJAsTNuvGuzs24UwcQ] another snapshot is currently running cannot delete"}],"type":"concurrent_snapshot_execution_exception","reason":"[full-policy-repo:snap-0zx3ektksugws-extubuza/6v6TuJAsTNuvGuzs24UwcQ] another snapshot is currently running cannot delete"},"status":503}
	at __randomizedtesting.SeedInfo.seed([A5EB03C9FA047982:AC32AEC6DADF21D7]:0)
	at org.elasticsearch.client.RestClient.convertResponse(RestClient.java:283)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:261)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:267)
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:235)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeSnapshots(ESRestTestCase.java:644)
	at org.elasticsearch.test.rest.ESRestTestCase.lambda$wipeCluster$4(ESRestTestCase.java:541)
	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:905)
	at org.elasticsearch.test.rest.ESRestTestCase.wipeCluster(ESRestTestCase.java:540)
	at org.elasticsearch.test.rest.ESRestTestCase.cleanUpCluster(ESRestTestCase.java:321)
```

https://gradle-enterprise.elastic.co/s/dfi2xbqejo4fu

Looks like it still fails on 7.x and 7.7 around 3-4 times a month. @dakrone should we mute those branches as well or are you looking for more infos?

**Comment by dakrone** [comment]
I believe this should be resolved now that #56911 has been merged, so I'm going to close it, we can re-open if it occurs again.
